src/components/ColorPicker.tsx

```diff
@@ -46,21 +56,35 @@ class ColorPicker extends PureComponent {
             action: 'Change Course Color',
         });
     };
-    updateColor = (color) => {
+    updateColor = (color: string) => {
         if (color !== this.props.color) {
             this.setState({ color: color });
         }
     };
 
     componentDidMount = () => {
+        let colorPickerId
+        if(this.props.isCustomEvent&&this.props.customEventID)
+            colorPickerId = this.props.customEventID.toString()
+        else if(this.props.sectionCode)
+            colorPickerId=this.props.sectionCode
+        else
+            throw new Error("Colorpicker custom component wasn't supplied a custom event id or a section code.")
         AppStore.registerColorPicker(
-            this.props.isCustomEvent ? this.props.customEventID : this.props.sectionCode,
+            colorPickerId,
             this.updateColor
         );
     };
     componentWillUnmount = () => {
+        let colorPickerId
+        if(this.props.isCustomEvent&&this.props.customEventID)
+            colorPickerId = this.props.customEventID.toString()
+        else if(this.props.sectionCode)
+            colorPickerId=this.props.sectionCode
+        else
+            throw new Error("Colorpicker custom component wasn't supplied a custom event id or a section code.")
```

Comment:
Semicolons missing

Reply:
Gonna let the autoformatter fix this in a separate PR